src/sagemaker/workflow/steps.py

```diff
@@ -808,7 +827,17 @@ def arguments(self) -> RequestType:
             request_dict = self.processor.sagemaker_session._get_process_request(**process_args)
 
         request_dict.pop("ProcessingJobName", None)
-        request_dict.pop("ExperimentConfig", None)
+
+        # only keep the trial component name
+        if request_dict.get("ExperimentConfig", {}).get("TrialComponentDisplayName"):
+            request_dict["ExperimentConfig"] = {
+                "TrialComponentDisplayName": request_dict["ExperimentConfig"][
+                    "TrialComponentDisplayName"
+                ]
+            }
+        else:
+            request_dict.pop("ExperimentConfig", None)
+
```

Comment:
same here, maybe worth having a outside utils entity?

Reply:
Yes. Refactored it out.